### PR TITLE
TR-9: create-spring-boot-oauth-implementation

### PR DESCRIPTION
**Pull request by YoofiTT96**
PR contains work done to create Spring Boot Application configured with Oauth2 JWT protection for all routes.
Documentation included to highlight relevant authentication urls and strategies

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (knocksto) [inline pom.xml:1]: This branch seems to be missing the main class that bootstraps the application. Let's also rename the package structure to com/turntablRides.

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      client:
+        registration:
+          google:
+            client-id: ${CLIENT_ID}
+            client-secret: ${CLIENT_SECRET}
+            scope:
+              - email
+              - profile
+
+  data:
```
Comment (knocksto) [inline src/main/resources/application.yml]: It'll be best to limit this feature to just google oauth wiithout adding mongoDB configurations unless that was required by your OAuth config.

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-test</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-oauth2-resource-server</artifactId>
+		</dependency>
+		<dependency>
```
Comment (knocksto) [inline pom.xml]: This PR should only contain dependencies relevant to OAuth configuration

File: src/main/java/com/ttrides/turntablRides/controllers/SampleController.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package com.ttrides.turntablRides.controllers;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+
+@RestController
+@RequestMapping("/api/v1/demo")
+public class SampleController {
```
Comment (knocksto) [inline src/main/java/com/ttrides/turntablRides/controllers/SampleController.java]: Can you rename this class to something more meaningful and relevant to it's purpose

**Comment by joesunesis** [comment]
Test Cases:
Accessing Web Resources without authorization.
Request-Response-Without-Header

Authorized User successfully accessing Web Resources.
Request-Response-With-Token-Header

**Review by knocksto — COMMENTED** [review]
Precede the PR title with the JiraId. Something like TR5: <PR-title-here>

File: pom.xml (hunk 3 above)
Comment (joesunesis) [reply pom.xml]: This dependency is very much relevant to OAuth configuration.
By the way it was either you or Yoofi who added it.

File: src/main/java/com/ttrides/turntablRides/controllers/SampleController.java (hunk 4 above)
Comment (joesunesis) [reply src/main/java/com/ttrides/turntablRides/controllers/SampleController.java]: Done

File: src/main/resources/application.yml (hunk 2 above)
Comment (joesunesis) [reply src/main/resources/application.yml]: Noted.

File: pom.xml (hunk 1 above)
Comment (joesunesis) [reply pom.xml:1]: I don't seem to understand what the comment is about.
Can you explain further?

Comment (knocksto) [reply pom.xml:1]: The above comment has been resolved so you can ignore

File: pom.xml (hunk 3 above)
Comment (joesunesis) [reply pom.xml]: Deleted irrelevant dependencies.
Comment resolved.

**Pull request merged**